tnnt/views.py

```diff
@@ -11,3 +26,310 @@ class AboutView(TemplateView):
+            self.leave_clan(request, player, ctx)
+
+        elif 'disband' in request.POST:
+            self.disband_clan(request, player, ctx)
+
+        elif 'join_clan' in request.POST:
+            self.join_clan(request, player, ctx)
+
+        elif 'rescind' in request.POST:
+            self.rescind_invite(request, player, ctx)
+
+        elif 'adminify' in request.POST:
+            self.make_admin(request, player, ctx)
+
+        elif 'kick' in request.POST:
+            self.kick_member(request, player, ctx)
+
+        return render(request, self.template_name, self.get_context_data(**ctx))
+
+    # Helper function triggered when a create_clan POST request comes in
+    def create_clan(self, request, player, ctx):
+        create_clan_form = CreateClanForm(request.POST)
+        new_clan_name = request.POST['clan_name']
+
+        if not create_clan_form.is_valid():
+            ctx['create_clan_form'] = create_clan_form
+            return
+
+        if Clan.objects.filter(name=new_clan_name).exists():
+            ctx['errmsg'] = 'That clan already exists'
+            return
+
+        if player.clan is not None:
+            ctx['errmsg'] = 'You are already in a clan'
+            return
+
+        # if we got here, we're good to create the clan, with the creator as its
+        # admin
+        newclan = Clan(name=new_clan_name)
+        newclan.save()
+        print('Player "%s" created clan "%s"' % (player.name, newclan.name))
+        player.clan = newclan
+        player.clan_admin = True
+        player.save()
+
+    # Helper function triggered when a invite-to-clan POST request comes in
+    def invite_to_clan(self, request, player, ctx):
+        invite_form = InviteMemberForm(request.POST)
+
+        if not invite_form.is_valid():
+            ctx['invite_member_form'] = invite_form
+            return
+
+        if player.clan is None:
+            # no impossible() >:(
+            # maybe someone's messing around with POST requests... >_>
+            impossible('Weird error? %s is clan admin but has no clan!' % (player.name))
+            ctx['errmsg'] = "You can't invite people without being in a clan! Please report this bug"
+            return
+
+        # only clan admins can invite players
+        if not player.clan_admin:
+            ctx['errmsg'] = "You can't invite people because you aren't a clan admin"
+            return
+
+        # if we got here, we're good to attempt the invite
+        try:
+            # retrieve player if exists in our database, if not
+            # but they are in dgl database, create the player in our
+            # database (we need to do this because we need to record
+            # them as having an invite)
+            invitee = hardfought_utils.find_player(request.POST['invitee'])
+            invitee.invites.add(player.clan)
+        except Player.DoesNotExist:
+            ctx['errmsg'] = 'No such player "' + request.POST['invitee'] + '" exists.'
```

Comment:
Not sure if they can do any actual damage with this, but I don't like unvalidated input being concatenated into the error message like this.